scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -1203,26 +1340,36 @@ def __get_log_config_for_container( self, cid, info, k8s_cache, base_attributes
             'container_name' : info['name'],
         }
 
+        rename_logfile = '/%s/%s.log' % (self._container_runtime, info['name'])
+
         k8s_info = info.get( 'k8s_info', {} )
 
         if k8s_info:
             pod_name = k8s_info.get('pod_name', 'invalid_pod')
             pod_namespace = k8s_info.get('pod_namespace', 'invalid_namespace')
             self._logger.log( scalyr_logging.DEBUG_LEVEL_1, "got k8s info for container %s, '%s/%s'" % (short_cid, pod_namespace, pod_name) )
-            pod = k8s_cache.pod( pod_namespace, pod_name )
+
+            rename_vars['pod_name'] = pod_name
+            rename_vars['namespace'] = pod_namespace
+            container_attributes['pod_name'] = pod_name
+            container_attributes['pod_namespace'] = pod_namespace
+
+            # get the cluster name
+            cluster_name = self.get_cluster_name( k8s_cache )
+            if self.__include_controller_info and cluster_name is not None:
```

Comment:
Question : Do we have an opportunity to set `_k8s_cn` here? I'm currently running profiling.6 where the env variable is set, and no problems.  However, I _think_ `self.__include_controller_info` is False which is why `_k8s_cn` is not set.

Can we safely eliminate the `self.__include_controller_info` check here and therefore set `_k8s_cn`??

Will that give us UI searcheability by cluster?

Reply:
Copying comments from slack for posterity:

If we pass `_k8s_cn` then we also need to pass `_k8s_dn` otherwise it gets automatically set to empty by the ingest pipeline.  This messes up the deployment names on the overview page and also leaves the search box without any autocomplete dropdowns.

My original code (version .5 I think) actually *did* include the `_k8s_cn`.  I took it out because of that problem.

We could try to discern the deployment name, and although we could make a decent guess, it adds too much uncertainty to the code - especially when we are hoping to pull in the actual deployment name from the API eventually.